hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -149,29 +149,28 @@ define([
                                 };
                               }
 
-                              return ContractService
-                                .fullDetails(revisionEntityIdObj.jobcontract_id)
-                                .then(function (results) {
-                                  var entity = {
-                                    contract: $scope.contract
-                                  },
-                                  contractRevisionIdObj = {
-                                    id: null,
-                                    jobcontract_id: contractId,
-                                    jobcontract_revision_id: results.details.jobcontract_revision_id
-                                  };
-
-                                  angular.extend(entity, angular.copy($scope.model));
-                                  angular.extend(entity.details, results.details);
-                                  angular.extend(entity.hour, results.hour || contractRevisionIdObj);
-                                  angular.extend(entity.pay, results.pay || contractRevisionIdObj);
-                                  angular.forEach(entity.leave, function (leaveType, leaveTypeId) {
-                                    angular.extend(leaveType, results.leave ? results.leave[leaveTypeId] || contractRevisionIdObj : contractRevisionIdObj);
-                                  });
-                                  angular.extend(entity.health, results.health || contractRevisionIdObj);
-                                  angular.extend(entity.pension, results.pension || contractRevisionIdObj);
-
-                                  return entity;
+                              return $q.all([
+                                ContractDetailsService.getOne({ jobcontract_revision_id: revisionEntityIdObj.details_revision_id}),
```

Comment:
could you have a local var for `{ jobcontract_revision_id: revisionEntityIdObj.details_revision_id}` and reuse it for multiple calls?